crates/neon-runtime/src/neon_task.h

```diff
@@ -0,0 +1,101 @@
+#ifndef NEON_TASK_H_
+#define NEON_TASK_H_
+
+#include <uv.h>
+#include "neon.h"
+#include "v8.h"
+
+namespace neon {
+
+class Task {
+public:
+  Task(v8::Isolate *isolate,
+       void *rust_task,
+       Neon_TaskPerformCallback perform,
+       Neon_TaskCompleteCallback complete,
+       v8::Local<v8::Function> callback)
+    : isolate_(isolate),
+      rust_task_(rust_task),
+      perform_(perform),
+      complete_(complete)
+  {
+    request_.data = this;
+    result_ = nullptr;
+    // Save the callback to be invoked when the task completes.
+    callback_.Reset(isolate, callback);
+    // Save the context (aka realm) to be used when invoking the callback.
+    context_.Reset(isolate, isolate->GetCurrentContext());
+  }
+
+  void execute() {
+    result_ = perform_(rust_task_);
+  }
+
+  void complete() {
+    // Ensure that we have all the proper scopes installed on the C++ stack before
+    // invoking the callback, and use the context (i.e. realm) we saved with the task.
+    v8::Isolate::Scope isolate_scope(isolate_);
+    v8::HandleScope handle_scope(isolate_);
+    v8::Local<v8::Context> context = v8::Local<v8::Context>::New(isolate_, context_);
+    v8::Context::Scope context_scope(context);
+
+    v8::Local<v8::Value> argv[2];
+
+    argv[0] = v8::Null(isolate_);
+    argv[1] = v8::Undefined(isolate_);
+
+    {
+      v8::TryCatch trycatch(isolate_);
```

Comment:
try/catch no longer deoptimizes in recent versions of V8.  But even if it did, it wouldn't matter because this is C++ code, not JS code - the performance pitfall only applied to JS functions with a try/catch statement.